APMrover2/AP_MotorsUGV.cpp

```diff
@@ -303,6 +312,12 @@ bool AP_MotorsUGV::have_skid_steering() const
     return false;
 }
 
+// true if the vehicle has a mainsail
+bool AP_MotorsUGV::has_sail() const
```

Comment:
I'm less sure about the cached bool idea although I see what you're saying.  It actually appears to be an issue with all the rover frame types - for example the have_skid_steering() check has the same issue.  Instead of fixing this I've raised an issue to capture the problem.  The issue is bigger than this PR.  https://github.com/ArduPilot/ardupilot/issues/9478